include/opt-sched/Scheduler/sched_region.h

```diff
@@ -77,11 +86,28 @@ class SchedRegion {
 
   // TODO(max): Document.
   virtual int CmputCostLwrBound() = 0;
+
+  virtual int cmputSpillCostLwrBound() = 0;
+
   // TODO(max): Document.
-  virtual InstCount UpdtOptmlSched(InstSchedule *crntSched,
-                                   LengthCostEnumerator *enumrtr) = 0;
+  virtual llvm::SmallVector<InstCount, 4>
+  UpdtOptmlSched(InstSchedule *crntSched, LengthCostEnumerator *enumrtr) = 0;
+
+  virtual llvm::SmallVector<InstCount, 4>
+  UpdtOptmlSchedFrstPss(InstSchedule *crntSched,
+                        LengthCostEnumerator *enumrtr) = 0;
+
+  virtual llvm::SmallVector<InstCount, 4>
+  UpdtOptmlSchedScndPss(InstSchedule *crntSched,
+                        LengthCostEnumerator *enumrtr) = 0;
+
+  virtual llvm::SmallVector<InstCount, 4>
+  UpdtOptmlSchedWghtd(InstSchedule *crntSched,
+                      LengthCostEnumerator *enumrtr) = 0;
```

Comment:
If there's a common reason why all of these functions' results have a small size of `4`, I'd prefer that the `4` be pulled out into a constant. If the reason can be explained with a comment on this constant, all the better, but that's not necessarily needed if the number is arbitrary.

Reply:
4 was just an arbitrary number that was big enough for returning what we need such as register pressure, schedule length, and possibly clustering in my project.